src/store/user-store.js

```diff
@@ -1,16 +1,48 @@
-import { makeAutoObservable } from "mobx";
+import { makeAutoObservable, flow } from "mobx";
 
 class UserStore {
   isLoggedIn = false;
 
+  userResponse = {};
+
   constructor(rootStore) {
     this.rootStore = rootStore;
-    makeAutoObservable(this);
+    makeAutoObservable(this, {
+      fetchUserInformation: flow
+    });
   }
 
   setIsLoggedIn(value) {
     this.isLoggedIn = value;
   }
+
+  setUserResponseData(data) {
+    this.userResponse = data;
+  }
+
+  *fetchUserInformation(token) {
+    const response = yield fetch('https://api.spotify.com/v1/me',{
+      headers: {
+        Authorization: `Bearer ${token}`
+      }
+    });
+
+    if (response.ok) {
+      const json = yield response.json();
+
+      this.setUserResponseData(json);
+    } else {
+      // TODO: make sure to do proper error handling
+    }
+  }
```

Comment:
Step 2: We need to create a new generator function (if you recall in the sync today, it's pretty much the exact same thing as `async/await` except we're using the `*` to depict that the function is asynchronous instead of `async`, and we're calling `yield` instead of `await`.

All we're doing here is moving the contents of the await function inside of `App.js` to this generator/flow function.

Note on line 33 we're calling `this.setUserResponseData(json)` to set the data